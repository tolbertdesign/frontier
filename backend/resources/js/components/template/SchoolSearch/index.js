import SchoolSearch from './SchoolSearch.vue'

export default SchoolSearch
