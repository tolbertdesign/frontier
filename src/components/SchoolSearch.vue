<template>
  <div class="school-search">
    <h2 class="text-30 mb-4">
      {{ $t('school_search_header') }}
    </h2>
    <div class="form-group mx-auto">
      <input
        :placeholder="$t('placeholder_school_search')"
        type="text"
        class="form-control text-16 mx-auto"
        name="school_search"
        required
        @input="search"
      >
      <ul
        v-show="schools.length"
        class="list-group"
      >
        <li
          v-for="(school, index) in schools"
          :key="index"
          :class="{ 'active': regCode === school.registration_code }"
          class="radio text-left text-16 list-group-item list-group-item-action"
          dusk="button"
          @click="selectSchool(school)"
        >
          <div class="font-weight-bold">
            {{ school.name }}
          </div>
          <div>{{ school.event_name }}</div>
          <div>{{ school.city }}, {{ school.state }}</div>
        </li>
      </ul>
    </div>

    <p class="text-12">
      {{ $t('search_help_1') }}
      <a
        class="text-primary font-weight-bold"
        href="#"
        data-target="#schoolSearchHelpModal"
        data-toggle="modal"
      >{{ $t('search_help_2') }}</a>
      {{ $t(' search_help_3') }}
    </p>
  </div>
</template>

<script>
export default {
  props: {
    errors: {
      type: Array,
      default: null,
    },
    searchUrl: {
      type: String,
      default: '',
    },
    schoolUrl: {
      type: String,
      default: '',
    },
  },
  data () {
    return {
      hideRegCode: true,
      regCode: '',
      schools: [],
    }
  },
  computed: {
  },
  methods: {
    search (event) {
      const searchTerm = event.target.value
      if (searchTerm.length >= 3) {
        axios.get(this.search_url + '/' + searchTerm)
          .then(response => {
            this.schools = response.data
          }).catch(error => {
            // TODO: LOGGLY
            // eslint-disable-next-line
            console.error(error)
          })
      } else {
        this.schools = []
      }
    },
    selectSchool (school) {
      window.location.href = this.schoolUrl + '/' + school.id
    },
  },
}
</script>

<style scoped>
ul{
    position: absolute;
    max-height:400px;
    overflow-y:scroll;
    overflow-x:hidden;
}
ul li {
    position: relative;
}
</style>
