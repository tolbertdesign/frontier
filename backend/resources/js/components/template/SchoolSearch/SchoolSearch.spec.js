import { shallowMount, createLocalVue } from '@vue/test-utils'
import SchoolSearch from './SchoolSearch.vue'

const localVue = createLocalVue()

const searchUrl = 'Sales'
const schoolUrl = 'School-URL'
const lang = {
  school_search_header: 'school search header',
  placeholder_school_search: 'placeholder school search',
  search_help_1: 'search help 1',
  search_help_2: 'search help 2',
  search_help_3: 'search help 3',
}

describe('SchoolSearch', () => {
  it('Snapshop test for SchoolSearch', () => {
    const wrapper = shallowMount(SchoolSearch, {
      localVue,
      propsData: {
        errors: [],
        searchUrl: searchUrl,
        schoolUrl: schoolUrl,
        lang: lang,
      },
    })
    expect(wrapper).toMatchSnapshot()
  })
})
