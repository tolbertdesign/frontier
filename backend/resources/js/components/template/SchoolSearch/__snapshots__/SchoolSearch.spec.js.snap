// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SchoolSearch Snapshop test for SchoolSearch 1`] = `
<div class="montserrat">
  <h2 class="text-30 mb-4">
    school search header
  </h2>
  <div class="form-group mx-auto"><input placeholder="placeholder school search" type="text" name="school_search" required="required" class="form-control text-16 mx-auto">
    <ul class="list-group" style="display: none;"></ul>
  </div>
  <p class="text-12">
    search help 1
    <a href="#" data-target="#schoolSearchHelpModal" data-toggle="modal" class="text-primary font-weight-bold">
      search help 2
    </a>
    search help 3
  </p>
</div>
`;
